www/app/api/chat/route.ts

```diff
@@ -131,6 +132,23 @@ async function fetchOpenRouter(type: string, messages: any[], payload: any) {
   }
 }
 
+async function verifyTurnstile(token: string, ip: string) {
+  const formData = new FormData();
+  formData.append('secret', process.env.TURNSTILE_SECRET_KEY as string);
+  formData.append('response', token);
+  formData.append('remoteip', ip);
+
+  const url = 'https://challenges.cloudflare.com/turnstile/v0/siteverify';
+  const result = await fetch(url, {
+    body: formData,
+    method: 'POST',
+  });
+
+  const json = await result.json();
+
+  return json.success;
+}
+
```

Comment:
So from briefly reading about cloudflare turnstiles it does look like a good option if we don't need a manual challenge. I was initially thinking to use the Supabase captcha, but this works.

The main thing we want to do is protect the authentication page to avoid bots and from reading about this we want to have a client side check that is verified on the server. 

That makes me think a dedicated server action for cloudflare would make sense that we call on the authentication page as well. We might want to add it to the `layout.tsx` so it's called on all pages?